tensorflow/python/keras/layers/dense_attention_test.py

```diff
@@ -757,6 +760,19 @@ def test_serialization(self):
     new_layer = dense_attention.AdditiveAttention.from_config(config)
     self.assertEqual(new_layer.use_scale, True)
 
+  def test_mixed_float16_policy(self):
```

Comment:
Need input from @reedwm to see if this is an appropriate way to test mixed precision behavior.

LGTM otherwise!

Reply:
Can you also add a test to [layer_correctness_test.py](https://github.com/tensorflow/tensorflow/blob/4133fef6917fbf973229a8a8047ba2056cd8b8ee/tensorflow/python/keras/mixed_precision/layer_correctness_test.py#L75) both for Attention and AdditiveAttention? Just need to add a parameter for both classes to the parameterized test. This automatically tests the outputs and gradients are the same between mixed precision and float32.

Also better to use the internal-only [policy_scope](https://github.com/tensorflow/tensorflow/blob/7d43911faa7bae5c56e3ffe4bd3c48e75ba37e5c/tensorflow/python/keras/mixed_precision/policy.py#L530) function to set the policy, which is equivalent to the try-finally statement used here

Reply:
Thanks @fchollet, the PR has been updated with comments addressed.